### PR TITLE
Improve Azure Pipelines support

### PR DESCRIPTION
**Pull request by dscho**
Let's backport a few changes from the review on the Git mailing list, really drop the `test_atexit` commits, drop the apt lock hack, parallelize the test suite on Windows, etc

Eventually all of this PR should be squashed into the next iteration of https://github.com/gitgitgadget/git/pull/31

### CODE REVIEW

File: t/helper/test-xml-encode.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+static const char *utf8_replace_character = "&#xfffd;";
+
+/*
+ * Encodes (possibly incorrect) UTF-8 on <stdin> to <stdout>, to be embedded
+ * in an XML file.
+ */
+int cmd__xml_encode(int argc, const char **argv)
+{
+	unsigned char buf[1024], tmp[3], *tmp2 = NULL;
```
Comment (jeffhostetler) [inline t/helper/test-xml-encode.c]: Shouldn't tmp be declared with [4] ?  the 0xF8 case needs 4 chars.

File: t/test-lib.sh [hunk 2]
```diff
@@ -891,15 +891,7 @@ xml_attr_encode () {
 	# We do not translate CR to &#x0d; because BSD sed does not handle
```
Comment (jeffhostetler) [inline t/test-lib.sh]: Is the CR comment stale now?  See line 56 in test-xml-encode.c

File: t/helper/test-xml-encode.c (hunk 1 above)
Comment (dscho) [reply t/helper/test-xml-encode.c]: Oooops!

File: t/test-lib.sh (hunk 2 above)
Comment (dscho) [reply t/test-lib.sh]: Ooops again!

**Comment by dscho** [comment]
This is now in sync with gitgitgadget/git#31 (which I just submitted as v3).

**Pull request merged**